contracts/StablecoinConverter.sol

```diff
@@ -209,14 +210,23 @@ contract StablecoinConverter is EpochTokenLocker {
                 volumes[i]
             );
         }
-        checkAndOverrideObjectiveValue(tokenConservation[0], batchIndex);
-        grantRewardToSolutionSubmitter();
+        checkAndOverrideObjectiveValue(uint(tokenConservation[0]), batchIndex);
+        grantRewardToSolutionSubmitter(uint(tokenConservation[0]) / 2);
         checkTokenConservation(tokenConservation);
         documentTrades(batchIndex, owners, orderIds, volumes, tokenIdsForPrice);
     }
 
-    function grantRewardToSolutionSubmitter() internal {
-        addBalanceAndPostponeWithdraw(msg.sender, tokenIdToAddressMap(0), uint(previousSolution.currentFeeCollected) / 2);
+    function getCurrentObjectiveValue() public view returns(uint) {
+        if (previousSolution.batchId == getCurrentStateIndex() - 1) {
```

Comment:
Silly question, wouldn't the `prevSolution.currentObjectiveValue` default to zero anyway if this condition weren't true? What is the reason we have the if statement where the second branch returns zero?

Reply:
No, previousSolution.objectiveValue just returns the objectiveValue for the current or last auction, which was dealt with. If it was not 0 in the last auction, and no one triggered the deletion of previousSolution object, then it would return a value > 0